app/services/change_offer.rb

```diff
@@ -1,16 +1,28 @@
 class ChangeOffer
   include ActiveModel::Validations
 
-  def initialize(actor:, application_choice:, course_option_id:)
+  attr_reader :application_choice, :course_option_id
+
+  validates_each :course_option_id do |record, attr, value|
+    record.errors.add(attr, :no_change) if value == record.application_choice.offered_option.id
+  end
+
+  def initialize(actor:, application_choice:, course_option_id:, offer_conditions: nil)
     @application_choice = application_choice
     @course_option_id = course_option_id
+    @offer_conditions = offer_conditions
     @auth = ProviderAuthorisation.new(actor: actor)
   end
 
-  def save!
+  def save
     @auth.assert_can_change_offer! application_choice: @application_choice, course_option_id: @course_option_id
```

Comment:
I've just noticed we roll together the right to change conditions with the right to change course. Maybe one to be aware of in future.

Reply:
Yes, see also my comment on the failing test. Changing an offer right now means changing the course_option_id and (optionally) the associated conditions and we don't support changing the conditions alone, hence the permission is accurate.